Python/Product/PythonTools/PythonTools/Environments/EnvironmentViewBase.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Data;
+using Microsoft.PythonTools.Interpreter;
+
+namespace Microsoft.PythonTools.Environments {
+    abstract class EnvironmentViewBase : DependencyObject, INotifyDataErrorInfo, IDisposable {
```

Comment:
Is it a view or a data context?

Reply:
The old classes were all named with the "View" suffix so that's why I kept it. If you have a suffix preference let me know.

Reply:
It's data context.